aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepTest.java

```diff
@@ -27,17 +25,15 @@
     /**
      * The simplest possible timeout step ever.
      */
-    @RandomlyFails("TODO sometimes the whole build just hangs (for longer than the 10s)")
     @Test
     public void basic() throws Exception {
         WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
         p.setDefinition(new CpsFlowDefinition(
-                "node { timeout(time:5, unit:'SECONDS') { sh 'sleep 10'; echo 'NotHere' } }"));
-        WorkflowRun b = assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0));
+                "node { timeout(time:5, unit:'SECONDS') { sleep 10; echo 'NotHere' } }"));
+        WorkflowRun b = assertBuildStatus(Result.ABORTED, p.scheduleBuild2(0));
```

Comment:
> standard Jenkins behaviour

FWIW the `build-timeout` plugin has a complex set of `BuildTimeOutOperation`s; the default seems to be to set the result to `ABORTED`. Unlike in a freestyle project, a workflow can customize timeout handling by just catching `InterruptedException` and doing something else, such as `error`.